src/Airship/Internal/Route.hs

```diff
@@ -80,14 +80,19 @@ star = Route [RestUnbound]
 newtype RoutingSpec m a = RoutingSpec { getRouter :: Writer [(Route, Resource m)] a }
     deriving (Functor, Applicative, Monad, MonadWriter [(Route, Resource m)])
 
-route :: [(Route, a)] -> [Text] -> a -> (a, (HashMap Text Text, [Text]))
-route routes pInfo resource404 = foldr' (matchRoute pInfo) (resource404, (mempty, mempty)) routes
+route :: [(Route, a)]
+      -> [Text]
+      -> Maybe (a, (HashMap Text Text, [Text]))
```

Comment:
When I see doubly-nested tuple types—note that we have an `(a, (b, [c]))` structure here, all wrapped in a Maybe—I find it hard to juggle the types in my head. Can we use a data structure for these various types, or, failing that, rub some `type` aliases on them?